<launch>
    <arg name="fixed_frame" default="world"/>

    <param name="robot_description" textfile="$(find hybrid_a_star)/model/car_model/default.urdf"/>
    <node pkg="joint_state_publisher" type="joint_state_publisher" name="joint_state_publisher"/>
    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher"/>

    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find hybrid_a_star)/rviz/hybrid_astar.rviz"/>

    <node pkg="tf" type="static_transform_publisher" name="link1_broadcaster"
          args="0 0 0 0 0 0 world map 100"/>

    <node pkg="hybrid_a_star" type="run_hybrid_astar" name="run_hybrid_astar" output="screen">
        <param name="planner/steering_angle" value="15.0"/>
        <param name="planner/steering_angle_discrete_num" value="1"/>
        <param name="planner/wheel_base" value="1.0"/>
        <param name="planner/segment_length" value="1.6"/>
        <param name="planner/segment_length_discrete_num" value="8"/>
        <param name="planner/steering_penalty" value="1.5"/>
        <param name="planner/reversing_penalty" value="2.0"/>
        <param name="planner/steering_change_penalty" value="2.0"/>
        <param name="planner/shot_distance" value="5.0"/>

        <!-- use or not use reference path cost -->
        <!-- <param name="planner/w_ref" value="0.03"/> -->
        <param name="planner/w_ref" value="0.0"/>

        <!-- use dubins or rs , dubins:0, rs:1 -->
        <param name="planner/use_dubins_or_rs" value="1"/>
    </node>
    
    <!-- <node pkg="hybrid_a_star" type="map_publish" name="map_publish" output="screen"/> -->
    <node pkg="hybrid_a_star" type="map_publish_y" name="map_publish" output="screen"/>

    <node pkg="hybrid_a_star" type="dy_obs_pub" name="dy_obs_pub" output="screen"/>

</launch>
